### DISCUSSION TITLE
v0.1.2

### DISCUSSION THREAD

**Discussion by josxha** [josxha/flutter-maplibre#135] [Announcements]
This release adds the last missing features for Android and Web, that are were
planned for now. The package will continue its ongoing efforts for stability
before iOS gets added as a supported platform.

### New Features

- Add `OfflineManager` for offline maps, cache management and bulk downloading.
- Add `PermissionsManager` to check and request location permissions.
- Add `WidgetLayer` to support widgets as Markers on the map.
- Add alternative Flutter platform view options for Texture Layer Hybrid Composition, Hybrid Composition, Virtual Display.
- Bump MapLibre Native on Android to version `11.6.+`.
- Add web-only controller functions `toScreenLocationSync()`, `toLngLatSync()`, `toScreenLocationsSync()`, `toLngLatsSync()`, `getMetersPerPixelAtLatitudeSync()` and `getVisibleRegionSync()`.
- Add `MapOptions.of(context)` and `MapOptions.maybeOf(context)`.
- Add `padding` and `alignment` parameters to the `MapScalebar` widget.

## Bug Fixes

- Fix WebAssembly builds.
- Remove unused `flutter_markdown` package.

## Misc

- Add unit tests, add Android integration tests.
- Use [codecov](https://app.codecov.io/gh/josxha/flutter-maplibre) to monitor test coverage.
- Improve file structure by using a `/platform` directory.

**Contributors:** @josxha
**Full Changelog:** [v0.1.1...v0.1.2](https://github.com/josxha/flutter-maplibre/compare/v0.1.1...v0.1.2)

<hr /><em>This discussion was created from the release <a href='https://github.com/josxha/flutter-maplibre/releases/tag/v0.1.2'>v0.1.2</a>.</em>

**Comment by ishafiul** [comment]
what about [flutter-maplibre-gl](https://github.com/maplibre/flutter-maplibre-gl)? Is its maintenance going to be shut down?

**Reply by josxha** [reply]
I'm not aware of any plans for flutter-maplibre-gl. 
The MapLibre team is not involved in flutter-maplibre-gl. The project is a [hosted MapLibre project](https://github.com/maplibre/maplibre/blob/main/PROJECT_TIERS.md). This means that it is not financial backed by the MapLibre organization and depends on voluntary maintainers. 

> A project that the MapLibre Organization does not allocate any resources to. The project is hosted in the MapLibre GitHub organization to facilitate collaboration between the people who are interested in the project.

The branch protection rules enforce pull requests to be reviewed before they can be merged. While this is good in general, it enforces more than one active collaborator at any time. The MapLibre team added two new collaborators to the probject after development stalled early this year, one of them was me. We were able to publish release 0.19.0 and 0.20.0 but then the other collaborator became inactive. The project would have had much potential but I was better off putting my efforts in a new package in the end.

Coming back to your question, flutter-maplibre-gl will probably sit around inactive for some more time (while recieving critical bug reports at a weekly basis).

**Reply by ishafiul** [reply]
Thanks for taking this step! I’m using MapLibre in my project and facing quite a few issues. I’m also new to the OSM world, so it’s a learning curve for me. and the community isn’t very active, which makes it more challenging. In the future, if my company decides to switch from MapLibre to another package, I can see that it might be difficult.

Anyway, I wish you the best of luck with this project. I’ll keep an eye on this repo and try to learn how everything works so I can contribute and build some great apps with this package.

Also, is there a Discord or Slack channel for discussions? And regarding iOS implementation—can we expect an iOS plugin soon, or will it take some time?

**Reply by josxha** [reply]
> I’ll keep an eye on this repo and try to learn how everything works so I can contribute and build some great apps with this package.

Awesome, thanks a lot. 

> Also, is there a Discord or Slack channel for discussions? And regarding iOS implementation—can we expect an iOS plugin soon, or will it take some time?

Yes, there is a slack channel called `#maplibre-flutter` on the [openstreetmap.us](https://slack.openstreetmap.us/) slack. It's always nice to get some discussions in there since it's not very active. But it's the place that the MapLibre team uses for their communitations as well, so it would be great to show them that MapLibre on Flutter indeed has users. 😄 

https://osmus.slack.com/archives/C05J5FMH7JN

**Reply by josxha** [reply]
> And regarding iOS implementation—can we expect an iOS plugin soon, or will it take some time?

Woops, missed to answer your last question. Yes, iOS is planned in the near future. Can't give an exact ETA though.

**Reply by smallTrogdor** [reply]
It's sad to hear about the path flutter-maplibre-gl is heading down ... 

Would it not have been wiser to request another (more active and committed) maintainer? Or were the problems with the project so deep?

I would honestly like to contribute in resolving more of the root issues and have a stable maplibre plugin for Flutter while waiting for flutter_map to become ready with Flutter GPU vector tile rendering 🚀 . But that may take some time (as you know I guess).

This leaves the Flutter community displaying VectorTiles with OpenSource tools in a very bad state unfortunately 😞 

**EDIT**

Found https://github.com/maplibre/flutter-maplibre-gl/discussions/503

This project makes me hopeful :)

**Reply by josxha** [reply]
> Would it not have been wiser to request another (more active and committed) maintainer? Or were the problems with the project so deep?

Looks like you already found my post where I describe some of the problems I see in flutter-maplibre-gl. 😄 I like that flutter-maplibre-gl has a relative good feature completeness but on the other there are the point I've listed in https://github.com/maplibre/flutter-maplibre-gl/discussions/503. Therefore, the code base itself is not really an argument for me to stick with the `flutter-maplibre-gl` project. I got into the package myself just a couple of weeks before. It was a "fork or join collaboration"-question back then (https://github.com/maplibre/flutter-maplibre-gl/issues/397). I joined as a maintainer so that the relatively small community around maplibre on flutter wouldn't become split. That said, it quickly became evident that a lot of people assumed that the package gets maintained by the MapLibre team and I had to explain several times the financial MapLibre structures. 

Don't get me wrong, I think what the people behind MapLibre achieved in fundraising and open source work while being relatively independent from companies is truely amazing. Nevertheles, I want to share a problem I think to see. All financial contributions are funneled towards their "core projects" MapLibre Native and MapLibre GL JS, this includes even funds from their hosted projects like `maplibre/flutter-maplibre-gl`. Not only does this cut off those projects from financial contributions, it makes it nearly impossible for those projects to find sponsors themselfs.
I would love to see that hosted projects could have their own buckets where they could collect and use their bugdets and even raise funds themselfs while collectively pushing the MapLibre brand. All hosted MapLibre projects should recieve a relatively small part of funding anyways compared to their core projects, but even this relatively little funding could already make a difference for R&D projects like [MapLibre-RS](https://github.com/maplibre/maplibre-rs) or packages like [flutter-maplibre-gl](https://github.com/maplibre/flutter-maplibre-gl).

(I'll link a somewhat related thread about MapLibre-RS on slack: https://osmus.slack.com/archives/C01G3D28DAB/p1731405967887099?thread_ts=1730856533.309349&cid=C01G3D28DAB)

> I would honestly like to contribute in resolving more of the root issues and have a stable maplibre plugin for Flutter [...]

As of today there are 62 bug reports for `flutter-maplibre-gl`. The (still ongoing) survey I launched recently states that 46% of users experienced app crashes using the package while 48% of users experienced problems with platform specific behaviors. I personally don't think that there are these hand full of root issues, but I'd be happy to be proven wrong.

> while waiting for flutter_map to become ready with Flutter GPU vector tile rendering 🚀 . But that may take some time (as you know I guess).

There was a similar discussion here: https://github.com/josxha/flutter-maplibre/discussions/114#discussioncomment-11134088

**Reply by smallTrogdor** [reply]
> Don't get me wrong, I think what the people behind MapLibre achieved in fundraising and open source work while being relatively independent from companies is truely amazing. Nevertheles, I want to share a problem I think to see. All financial contributions are funneled towards their "core projects" MapLibre Native and MapLibre GL JS, this includes even funds from their hosted projects like maplibre/flutter-maplibre-gl. Not only does this cut off those projects from financial contributions, it makes it nearly impossible for those projects to find sponsors themselfs.

Yes, I guess that is a fundamental problem of the MapLibre organization.

> As of today there are 62 bug reports for flutter-maplibre-gl. The (still ongoing) survey I launched recently states that 46% of users experienced app crashes using the package while 48% of users experienced problems with platform specific behaviors. I personally don't think that there are these hand full of root issues, but I'd be happy to be proven wrong.

It's true, this does not seem like that much of a big deal, but having gone through the code base I see more issues. Furthermore, the lack of good test makes it very hard for unwanted behavior changes imo. Also, I tried this plugin on an Android device and it works already more stable than the flutter-maplibre-gl in my opinion. So you have already put a pole into the ground ...
